### PR TITLE
fix(Global Search/Tagging): regenerated services with latest API definition

### PR DESCRIPTION
**Pull request by fabrizio-leoni**
Signed-off-by: Fabrizio Leoni <fabrizio_leoni@it.ibm.com>

## PR summary

SDK regenerated API definition starting from https://test.cloud.ibm.com/apidocs/tagging https://test.cloud.ibm.com/apidocs/search

## PR Checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## Does this PR introduce a breaking change?    
- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by fabrizio-leoni** [comment]
@padamstx

**Review by padamstx — APPROVED** [review]
LGTM

File: globalsearchv2/global_search_v2.go [hunk 1]
```diff
@@ -403,13 +431,13 @@ func UnmarshalResultItem(m map[string]json.RawMessage, result interface{}) (err
 // ScanResult : The search scan response.
 type ScanResult struct {
 	// The search cursor to use on all calls after the first one.
-	SearchCursor *string `json:"search_cursor" validate:"required"`
+	SearchCursor *string `json:"search_cursor,omitempty"`
 
 	// Value of the limit parameter specified by the user.
-	Limit *int64 `json:"limit,omitempty"`
+	Limit *float64 `json:"limit" validate:"required"`
```
Comment (padamstx) [inline globalsearchv2/global_search_v2.go:437]: The "limit" property of the ScanResult schema should most likely be defined as "integer" instead of "number". Apparently, back when this service was last re-generated (by @damianovesperini on 6/10/2021) the property was in fact defined as "integer" which is why the current version of the code is using "int64".  It looks like search.yaml was updated in [this commit](https://github.ibm.com/cloud-api-docs/tagging/commit/a89b4779a1496f75528229764fe497459940f16b) to use "number" instead of "integer".  I think this should be changed back to "integer" in the API definition.

**Review by padamstx — CHANGES_REQUESTED** [review]
We need to address the problem related to the ScanResult.limit property

File: globalsearchv2/global_search_v2.go [hunk 2]
```diff
@@ -34,22 +34,19 @@ import (
...
 // properties of a resource include tags and system properties. Both properties are defined in an IBM Cloud billing
 // account, and span across many regions.
 //
-// Version: 2.0.1
+// API Version: 2.0.1
 type GlobalSearchV2 struct {
 	Service *core.BaseService
 }
 
-// DefaultServiceURL is the default URL to make service requests to.
```
Comment (padamstx) [inline globalsearchv2/global_search_v2.go:50]: This default service URL was removed because the "servers" field was removed from the API definition. The "servers" field defines the default service endpoint to be used by users of the service. Why was that removed?

Comment (fabrizio-leoni) [reply globalsearchv2/global_search_v2.go:50]: image

**Comment by fabrizio-leoni** [comment]
image
image

File: globalsearchv2/global_search_v2.go (hunk 1 above)
Comment (fabrizio-leoni) [reply globalsearchv2/global_search_v2.go:437]: At this question can replay Michele Crudele.

**Comment by fabrizio-leoni** [comment]
image

File: globalsearchv2/global_search_v2.go (hunk 2 above)
Comment (fabrizio-leoni) [reply globalsearchv2/global_search_v2.go:50]: At this question can replay Michele Crudele.

File: globalsearchv2/global_search_v2.go (hunk 1 above)
Comment (fabrizio-leoni) [reply globalsearchv2/global_search_v2.go:437]: It will be fixed

File: globalsearchv2/global_search_v2.go (hunk 2 above)
Comment (mkrudele) [reply globalsearchv2/global_search_v2.go:50]: @padamstx I believe the team removed the `servers` from the definitions because we use the swaggers to point to our internal test environments, which we cannot reference publicly. Is that a problem not setting the `servers`?

Comment (padamstx) [reply globalsearchv2/global_search_v2.go:50]: Yes, in general one should be using the production version of the API definition when generating code for the SDKs.  Is it a problem to define the production endpoint in the servers field within all three branches (master, test, production)?

Comment (fabrizio-leoni) [reply globalsearchv2/global_search_v2.go:50]: done

Comment (fabrizio-leoni) [reply globalsearchv2/global_search_v2.go:50]: done

File: globalsearchv2/global_search_v2.go (hunk 1 above)
Comment (padamstx) [reply globalsearchv2/global_search_v2.go:437]: @fabrizio-leoni When the API definition is fixed for this problem and the endpoint URL issue, you'll need to regen the SDK code one more time, then update the PR with the changes.  Also, be sure to pull in the latest changes from the main branch as well.

**Review by padamstx — APPROVED** [review]
LGTM
